docs: new tutorials (#3292)
refinedev/refine: documentation/docs/tutorial/3-understanding-resources/0-intro.md

To add a `resource` to our app, we need use `resources` prop of `<Refine/>` component. This prop accepts an array of objects. Each object represents a resource.

```tsx title="src/App.tsx"
import { Refine } from "@pankod/refine-core";
import routerProvider from "@pankod/refine-react-router-v6";
import dataProvider from "@pankod/refine-simple-rest";

const App: React.FC = () => {
    return (
        <Refine
            routerProvider={routerProvider}
            dataProvider={dataProvider("https://api.fake-rest.refine.dev")}
            //highlight-start
            resources={[
                {
                    name: "products",
                    list: () => <div>Products List Page</div>,
                    show: <div>Product Detail Page</div>,
                    create: <div>Product Create Page</div>,
                    edit: <div>Product Edit Page</div>,
                },
            ]}
            //highlight-end
        />
    );
};

export default App;
```

> In the above example, we have added a `resource` with name `products`. This resource will have 5 pages that assigned to `list`, `show`, `create`, and `edit` properties. **refine** link the pages with routes for us. When we define `create`, clone page will be generated automatically. We can access these pages by visiting the following routes:

Reviewer: ```suggestion
In the above example, we have added a `resource` with name `products`. This resource will have 5 pages that assigned to `list`, `show`, `create`, and `edit` properties. **refine** link the pages with routes for us.  

When we define `create`, clone page will be generated automatically. We can access these pages by visiting the following routes:
```

---
id: index
title: Resources
tutorial:
    order: 0
    prev: false
    next: tutorial/adding-crud-pages/{preferredUI}/index
---

:::info Remember

In the [Unit 3.2](#), we have defined a resource to create our CRUD pages with the Inferencer component. However, we have not explained how it works. In this unit, we will explain the `resources` prop of the `<Refine />` component using the mock components.

> 

Reviewer: I think in the resources unit, it would be nice if we can talk about <Refine> component in thr App.tsx a little bit. Maybe something like, it's heart of the refine apps, dataproviders, router providers and other important features need to register here.
Maybe giving link to the <Refine> property docs..
@salihozdemir

:::

`resources` are the main building block of **refine** app. Each resource allows you to create new routes of CRUD pages in your app.

> 

Reviewer: In the context of a CRUD  application, a resource typically refers to a data entity that can be created, read, updated, or deleted. For example, a resource could be a user account, a blog post, a product in an online store, or any other piece of data that can be managed by the CRUD app.  
Maybe its better if we put general explanation of what does "resource" mean. @salihozdemir

Reviewer: Dİd we put general explanation for "resource" term? @salihozdemir

Let's see what's going on under the hood when we add a `resources` to our app and learn `resources` term for who are new to **refine**.

> To add `resource` to our app, we must use `resources` prop of `<Refine />` component. This prop accepts an array of objects. Each object represents a resource.

Reviewer: ```suggestion
To add  a `resource` to our app, we need use `resources` prop of `<Refine />` component. This prop accepts an array of objects. Each object represents a resource.
```

```tsx title="src/App.tsx"
import { Refine } from "@pankod/refine-core";
import routerProvider from "@pankod/refine-react-router-v6";
import dataProvider from "@pankod/refine-simple-rest";

const App: React.FC = () => {
    return (
        <Refine
            routerProvider={routerProvider}
            dataProvider={dataProvider("https://api.fake-rest.refine.dev")}
            //highlight-start
            resources={[
                {
                    name: "posts",
                    list: () => <div>Posts List Page</div>,
                    show: <div>Post Detail Page</div>,
                    create: <div>Post Create Page</div>,
                    edit: <div>Post Edit Page</div>,
                },
            ]}
            //highlight-end
        />
    );
};

export default App;
```

> In the above example, we have added a `resource` with name `posts`. This resource has 5 pages. `list`, `show`, `create`, `clone` and `edit` pages and **refine** link the pages with routes for us. We can access these pages by visiting the following routes:

Reviewer: ```suggestion
In the above example, we have added a `resource` with name `posts`. This resource will have 5 pages that assigned to `list`, `show`, `create`, `clone` and `edit`  properties. **refine** link the pages with routes for us. We can access these pages by visiting the following routes:
```
